pkg/cmd/pipeline/describe.go

```diff
@@ -0,0 +1,135 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pipeline
+
+import (
+	"fmt"
+	"io"
+	"text/tabwriter"
+
+	"github.com/spf13/cobra"
+	"github.com/tektoncd/cli/pkg/cli"
+	"github.com/tektoncd/cli/pkg/formatted"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	cliopts "k8s.io/cli-runtime/pkg/genericclioptions"
+)
+
+const (
+	resHeader  = "\nRESOURCE\nNAME\tTYPE\n"
+	resBody    = "%s\t%s\n"
+	taskHeader = "\nTASK\nNAME\tTASKREF\tRUNAFTER\n"
+	taskBody   = "%s\t%s\t%v\n"
+	runsHeader = "\nRUN\nNAME\tSTARTED\tDURATION\tSTATUS\n"
+	runsBody   = "%s\t%s\t%s\t%s\n"
+)
+
+var pname string
+
+func describeCommand(p cli.Params) *cobra.Command {
+	f := cliopts.NewPrintFlags("describe")
+
+	c := &cobra.Command{
+		Use:     "describe",
+		Aliases: []string{"desc"},
+		Short:   "Describes a pipeline in a namespace",
+		Long:    ``,
+		Args:    cobra.MinimumNArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			pname = args[0]
+			return printPipelineDescription(cmd.OutOrStdout(), p)
+		},
+	}
+	f.AddFlags(c)
+	return c
+}
+
+func printPipelineDescription(out io.Writer, p cli.Params) error {
+	cs, err := p.Clients()
+	if err != nil {
+		return err
+	}
+
+	c := cs.Tekton.TektonV1alpha1().Pipelines(p.Namespace())
+	ds, err := c.Get(pname, metav1.GetOptions{})
+	if err != nil {
+		fmt.Fprintln(out, "Error :", err)
+		return nil
+	}
+
+	w := tabwriter.NewWriter(out, 0, 5, 3, ' ', tabwriter.TabIndent)
+	fmt.Fprintf(out, "\nNAME : %s\n", pname)
+	printResources(w, ds)
+	printTasks(w, ds)
+	printRuns(w, cs, p)
+
+	return w.Flush()
+}
+
+func printResources(w *tabwriter.Writer, ds *v1alpha1.Pipeline) {
+	fmt.Fprintf(w, resHeader)
+	if len(ds.Spec.Resources) == 0 {
+		fmt.Fprintf(w, "No resources")
+		return
+	}
+	for _, val := range ds.Spec.Resources {
+		fmt.Fprintf(w, resBody,
+			val.Name,
+			val.Type,
+		)
+
+	}
+}
+
+func printTasks(w *tabwriter.Writer, ds *v1alpha1.Pipeline) {
+	fmt.Fprintf(w, taskHeader)
+	if len(ds.Spec.Tasks) == 0 {
+		fmt.Fprintf(w, "No tasks")
+		return
+	}
+	for _, val := range ds.Spec.Tasks {
+		fmt.Fprintf(w, taskBody,
+			val.Name,
+			val.TaskRef.Name,
+			val.RunAfter,
+		)
+
+	}
+}
+
+func printRuns(w *tabwriter.Writer, cs *cli.Clients, p cli.Params) error {
+	fmt.Fprintf(w, runsHeader)
+	runs, err := cs.Tekton.TektonV1alpha1().PipelineRuns(p.Namespace()).List(metav1.ListOptions{})
+	if err != nil {
+		return err
```

Comment:
I think you were right, instead of returning an error we can print the error message only.